src/EnergyPlus/HeatRecovery.hh

```diff
@@ -94,413 +81,318 @@ namespace HeatRecovery {
-        Real64 T_MaxRegenAirInRelHum; // max allowable regen inlet air relative humidity [%]
-        Real64 T_MinProcAirInRelHum;  // min allowable process inlet air relative humidity [%]
-        Real64 T_MaxProcAirInRelHum;  // max allowable process inlet air relative humidity [%]
-        // regeneration outlet humidity ratio equation coefficients and limits
-        Real64 C1;                    // constant coeff for outlet regen humidity ratio equation
-        Real64 C2;                    // regen inlet humrat coeff for outlet regen humidity ratio eq
-        Real64 C3;                    // regen inlet temp coeff for outlet regen humidity ratio equation
-        Real64 C4;                    // (regen in humrat/regen in temp) coeff for outlet regen humrat eq
-        Real64 C5;                    // process inlet humrat coeff for outlet regen humidity ratio eq
-        Real64 C6;                    // process inlet temp coeff for outlet regen humidity ratio eq
-        Real64 C7;                    // (proc in humrat/proc in temp) coeff for outlet regen humrat eq
-        Real64 C8;                    // process, regen face velocity coeff for outlet regen humrat eq
-        Real64 H_MinRegenAirInTemp;   // min allowable regen inlet air temperature [C]
-        Real64 H_MaxRegenAirInTemp;   // max allowable regen inlet air temperature [C]
-        Real64 H_MinRegenAirInHumRat; // min allowable regen inlet air humidity ratio [kg water / kg air]
-        Real64 H_MaxRegenAirInHumRat; // max allowable regen inlet air humidity ratio [kg water / kg air]
-        Real64 H_MinProcAirInTemp;    // min allowable process inlet air temperature [C]
-        Real64 H_MaxProcAirInTemp;    // max allowable process inlet air temperature [C]
-        Real64 H_MinProcAirInHumRat;  // min allowable process inlet air humidity ratio [kg water/kg air]
-        Real64 H_MaxProcAirInHumRat;  // max allowable process inlet air humidity ratio [kg water/kg air]
-        Real64 H_MinFaceVel;          // min allowable process, regen face velocity [m/s]
-        Real64 H_MaxFaceVel;          // max allowable process, regen face velocity [m/s]
-        Real64 MinRegenAirOutHumRat;  // min allowable regen outlet air temperature [C]
-        Real64 MaxRegenAirOutHumRat;  // max allowable regen outlet air temperature [C]
-        Real64 H_MinRegenAirInRelHum; // min allowable regen inlet air relative humidity [%]
-        Real64 H_MaxRegenAirInRelHum; // max allowable regen inlet air relative humidity [%]
-        Real64 H_MinProcAirInRelHum;  // min allowable process inlet air relative humidity [%]
-        Real64 H_MaxProcAirInRelHum;  // max allowable process inlet air relative humidity [%]
-        // for model bound checking
-        // regen inlet relative humidity for temperature equation
-        bool PrintRegenInRelHumTempMess;      // - flag to print regen in RH error message for temp eq
```

Comment:
This pattern of `bool`, `int`, `int`, `string`, `string`, `string`, `Real64` was repeated over, and over, and over, and over here, just to track recurring warnings.  I collected them into a meaningful structure and replaced them all.  A huge pain, but it's so much cleaner now.  A future work should expand this to all components utilizing the recurring warning functionality so that they can all just utilize instances of this struct and it could easily eliminate hundreds/thousands of lines of code.